### DISCUSSION TITLE
Save a layer as a separate image

### DISCUSSION THREAD

**Discussion by wwangxinhao** [pymupdf/PyMuPDF#2805] [Q&A]
def print_xobjects_info(pdf_path):
    doc = fitz.open(pdf_path)
    
    # 获取文档中的所有图层
    layers = doc.get_ocgs()
    print("All Layers:", layers)
    
    # 获取当前文档中所有图层的信息
    all_layers_info = doc.get_layer()
    print("All Layers Info:", all_layers_info)
    
    #可见性设置
    # 设置图层可见性(可见性设置是管用的，但是并没有删除图层)
    doc.set_layer(-1, on=[21, 25, 223, 267, 227, 13, 235], off=[256])
    # # 保存修改
    # doc.save("modified_document.pdf")
    
    # 再次获取图层信息，检查是否设置成功
    updated_layers_info = doc.get_layer()
    print("Updated Layers Info:", updated_layers_info)
    
    layers = doc.get_ocgs()
    print("All Layers:", layers)
    
    #获取图像信息
    page_iamges = doc.get_page_xobjects(0)
    print("page_iamges:", page_iamges)
    
 
    # 加载第一页
    page = doc.load_page(0)
    
    
    # 获取页面 pixmap
    pixmap = page.get_pixmap()

    # 将 Pixmap 转换为 PIL Image
    img = Image.frombytes("RGB", [pixmap.width, pixmap.height], pixmap.samples)

    # 保存图像，文件名以图层名命名
    img_filename = f"page_0.png"
    img.save(img_filename)
    print(f"Saved: {img_filename}")
    
    doc.close()

# Example usage
pdf_path = 'test/test.pdf'
print_xobjects_info(pdf_path)

this is mycode， doc.set_layer  is work ,but i dont know how to save a separate image, for example layer 21

**Comment by JorjMcKie** [comment]
1. You do not need Pillow to save PNG or JPEG images. Simply use `pix.save()` giving the filename the appropriate extemsion.
2. When you activate a layer, then all content extraction and Pixmap creation will react to this.
